### DISCUSSION TITLE
Plans For Version 2

### DISCUSSION THREAD

**Discussion by Ratstail91** [krgamestudios/Toy#119] [Announcements]
I've spent some time away from this project, which has given me a useful perspective.

I've determined that the scope's storage system is inadequate; I need to rework things into a stack-like system for local variable storage. I haven't decided on the exact details yet, but I do know that this will require a major version bump, as well as a major rewrite. My hope is that, by reworking storage and the literal cache system, I can speed up the performance enough to make embedding this into a game engine a viable use-case.

Therefore, version 1.x is now considered in maintenance mode - no new features will be added at this point, and all efforts are to be directed to the new version.

There's no expected start or finish dates, but I'm hoping to clear up some other things so I can tackle this soon.

**Comment by hiperiondev** [comment]
Greetings... how long has it been!
I'm working on a new VM, maybe you can use it for the new version: https://github.com/hiperiondev/stack_vm
It is based on the VM of another scripting language (https://github.com/goodpaul6/Tiny)
It has frame and heap management.
It also has a disassembler and a complete assembler with directive management, it is also easily extensible

**Reply by Ratstail91** [reply]
Hi, good to see you!

Right now, I'm not focused on Toy, but I know that I *will* come back to it one day - possibly later in the year. For now, I wanna do something nice and easy, because life has been too full on lately.

Thanks for these references! I'll have a close look at them when I come back to this.

BTW, if you use discord, I have a server for my [game studio here](https://discord.gg/bswm3ux4N7), and I'd love it if you wanted to post about your progress there. It's not exclusively for games, as general coding is welcome too - I'm trying to build a nice community.
